tests/models/test_docker_utils.py

```diff
@@ -0,0 +1,319 @@
+            ),
+        ),
+        (
+            dict(
+                HTTP_PROXY="http://foo:8080",
+                NO_PROXY="qux",
+                http_proxy_protocol="http",
+                http_proxy_host="foo",
+                http_proxy_port="8080",
+                non_proxy_hosts="qux",
+            ),
+        ),
+        (
+            dict(
+                HTTP_PROXY="http://foo:8080",
+                NO_PROXY="qux,quux",
+                http_proxy_protocol="http",
+                http_proxy_host="foo",
+                http_proxy_port="8080",
+                non_proxy_hosts="qux|quux",
+            ),
+        ),
+        (
+            dict(
+                HTTP_PROXY="http://bar@foo:8080",
+                http_proxy_protocol="http",
+                http_proxy_host="foo",
+                http_proxy_port="8080",
+                http_proxy_user="bar",
+            ),
+        ),
+        (
+            dict(
+                HTTP_PROXY="http://bar:baz@foo:8080",
+                http_proxy_protocol="http",
+                http_proxy_host="foo",
+                http_proxy_port="8080",
+                http_proxy_user="bar",
+                http_proxy_password="baz",
+            ),
+        ),
+        (
+            dict(
+                HTTPS_PROXY="http://foo:8080",
+                https_proxy_protocol="http",
+                https_proxy_host="foo",
+                https_proxy_port="8080",
+            ),
+        ),
+        (
+            dict(
+                HTTPS_PROXY="http://foo:8080",
+                NO_PROXY="qux",
+                https_proxy_protocol="http",
+                https_proxy_host="foo",
+                https_proxy_port="8080",
+                non_proxy_hosts="qux",
+            ),
+        ),
+        (
+            dict(
+                HTTPS_PROXY="http://foo:8080",
+                NO_PROXY="qux,quux",
+                https_proxy_protocol="http",
+                https_proxy_host="foo",
+                https_proxy_port="8080",
+                non_proxy_hosts="qux|quux",
+            ),
+        ),
+        (
+            dict(
+                HTTPS_PROXY="http://bar@foo:8080",
+                https_proxy_protocol="http",
+                https_proxy_host="foo",
+                https_proxy_port="8080",
+                https_proxy_user="bar",
+            ),
+        ),
+        (
+            dict(
+                HTTPS_PROXY="http://bar:baz@foo:8080",
+                https_proxy_protocol="http",
+                https_proxy_host="foo",
+                https_proxy_port="8080",
+                https_proxy_user="bar",
+                https_proxy_password="baz",
+            ),
+        ),
+    ],
+)
+def test_get_maven_settings(kwargs):
+    # Multiprocessing is used here so we can set env vars,
+    # without risking breaking the actual environment,
+    # and lets us test the logic of using both cases of variables
+    q = multiprocessing.Queue()
```

Comment:
We can use pytest's monkeypatch fixture to mock environment variables

Reply:
Wasn't aware of that option, thanks. TIL.